app/routes/articles.js

```diff
@@ -198,62 +194,65 @@ async function getRecommendedArticles(category) {
+    } catch(err) {
+        req.flash('error', 'Oops! Something went wrong!')
+        req.log('Article EDIT Route:', err)
+        return res.redirect('/')
+    }
 })
 
 //UPDATE Route
-router.put('/:link', checkArticleOwnership, validation(updateArticle, BODY), (req, res) => {
+
+router.put('/:link', checkArticleOwnership, validation(updateArticle, BODY), async (req, res) => {
     if(req.params.link === undefined) {
         req.flash('error', 'Oops! Something went wrong!')
         req.log('Article UPDATE Route:', req.body)
         return res.redirect('/')
     }
-
-    Article.findOneAndUpdate({link: req.params.link}, {$set: req.body}, {runValidators: true}, (err, article) => {
-        if(err) {
-            if(err?.errors?.properties?.type === 'minlength' || err?.errors?.properties?.type === 'maxlength') {
-                return res.render('error', {code: '401', msg: 'Invalid input length.'})
-            }
-
-            if(err._message === 'Article validation failed') {
-                req.flash('error', 'Invalid input lengths, please try again.')
-                return res.redirect(`back`)
-            }
-            
-            req.flash('error', 'Oops! Something went wrong!')
-            req.log('Article UPDATE Route:', err)
-            return res.redirect('/')
-        }
+    try {
+        const article = await Article.findOneAndUpdate({link: req.params.link}, {$set: req.body}, {runValidators: true}).populate('author').exec()
         if(article.title !== req.body.title) article.link = encodeURIComponent(req.body.title.replace(SPACES, DASH))
         req.flash('success', 'Successfully updated your article!')
-        res.redirect('/articles/' + req.params.link)
-    })
+        return res.redirect('/articles/' + req.params.link)
+    } catch(err) {
+        if(err?.errors?.properties?.type === 'minlength' || err?.errors?.properties?.type === 'maxlength') {
+            return res.render('error', {code: '401', msg: 'Invalid input length.'})
+        }
+
+        if(err._message === 'Article validation failed') {
+            req.flash('error', 'Invalid input lengths, please try again.')
+            return res.redirect(`back`)
+        }
+        
+        req.flash('error', 'Oops! Something went wrong!')
+        req.log('Article UPDATE Route:', err)
+        return res.redirect('/')
+    }
 })
 
 //DELETE Article Route
-router.delete('/:link', checkArticleOwnership, (req, res) => {
+router.delete('/:link', checkArticleOwnership, async (req, res) => {
     if(!req.params.link) return res.render('error', {code: '404', msg: 'Invalid Article Link'})
-   
-    Article.deleteOne({link: req.params.link}, err => {
-        if(err) return res.render('error', {code: '500', msg: 'Internal Database Error'})
-
-        Counter.deleteOne({articleLink: req.params.link}).exec()
+    
+    try {
+        await Article.deleteOne({link: req.params.link}).exec()
+        await Counter.deleteOne({articleLink: req.params.link}).exec()
```

Comment:
Is it possible for two articles to have the same link?
i.e  articles with titles `How to hack node` and `How to "hack" node`

Reply:
Article titles are alphanumeric that allow dashes and spaces, it shouldn't be possible, link must be unique.